.config/gitpod/Dockerfile

```diff
@@ -38,23 +36,37 @@ RUN PYTHON_VERSION=$(cut -d '=' -f 2 /tmp/python_version.config) && \
         vim \
         wget \
     && python3 -m pip install --no-cache-dir --upgrade pip \
-    && python3 -m pip install --no-cache-dir virtualenv \
-    # Install Docker
-    && curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg \
+    && python3 -m pip install --no-cache-dir virtualenv
+
+# Install Docker
+FROM builder as docker-installer
+
+RUN curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg \
     && echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu \
     $(lsb_release -cs) stable" | tee /etc/apt/sources.list.d/docker.list > /dev/null \
     && apt-get update \
-    && apt-get install -y --no-install-recommends docker-ce docker-ce-cli containerd.io \
-    # Install npm and ungit using the versions specified in the variables
-    && curl -fsSL https://deb.nodesource.com/setup_20.x |bash - \
+    && apt-get install -y --no-install-recommends docker-ce docker-ce-cli containerd.io
+
+# Install npm and ungit
+FROM docker-installer as npm-installer
+
+RUN curl -fsSL https://deb.nodesource.com/setup_20.x | bash - \
     && apt-get install  -y --no-install-recommends nodejs \
     && npm install -g npm@${NPM_VERSION} \
     && npm install -g \
-        ungit@${UNGIT_VERSION} \
-    # Clean
-    && apt-get clean && rm -rf /var/lib/apt/lists/* tmp/*
+        ungit@${UNGIT_VERSION}
+
+# Cleanup
+FROM npm-installer as cleanup
+
+RUN apt-get clean && rm -rf /var/lib/apt/lists/* tmp/* 
+
+# Final stage: Setup final image and user
+FROM ubuntu:jammy
```

Comment:
To avoid including the layers from the ubuntu:jammy image in our final image and considering that you're copying the root from the last stage instead of the ubuntu:jammy root, it would be preferable to start from the scratch image.

`FROM scratch`

https://hub.docker.com/_/scratch

Reply:
@ThomasSanson 

I have just made the modification "FROM scratch" and after running "make tests" and "make docker-tests", I did not receive any error messages.

You can find the modifications in this commit: dbb88523c77c8201d1f7b5975d292e4f64d0eb68